src/onnxruntime.cc

```diff
@@ -1598,6 +1600,13 @@ ModelInstanceState::ProcessRequests(
             memory_type_id = 0;
           }
         }
+
+        // If the instance type is not KIND_GPU, set the memory type to CPU.
+        if (Kind() != TRITONSERVER_INSTANCEGROUPKIND_GPU) {
```

Comment:
Can you check if `cuda_allocator_info_` is initialized instead? (cuda_allocator_info_ == nullptr). That is the direct cause of can't put output to device memory, and I would imagine the use case where the model is deployed on CPU but have `cuda_allocator_info_` set so that ORT will place the output to device memory if desired.

Reply:
Sounds good. Fixed.